lib/dynflow/testing/managed_clock.rb

```diff
@@ -26,6 +26,18 @@ def progress
         end
       end
 
+      def skip_first?(ignored_subjects)
+        item = @pending_pings.first
+        item && item.what.respond_to?(:value) && ignored_subjects.include?(item.what.value)
+      end
+
+      def progress_all(ignored_subjects)
+        until @pending_pings.empty? || !skip_first?(ignored_subjects)
```

Comment:
I believe the `ignored_subjects` should be a parameter of progress, and in case should be ignored, just do `shift` but not actually applying it. I'm not sure this code does what it should, in terms of just dropping the events that should be ignored.

Reply:
it does work (it just also apply's ignored pings, which i can see maybe is not desired).  will fix

Reply:
updated!